content/article/why-i-hate-json-and-you-can-too-.md

```diff
@@ -0,0 +1,177 @@
+
+  {
+    "title"       : "Why I Hate JSON (and you can, too)",
+    "authors"     : ["felipe-gasper"],
+    "date"        : "2019-11-10T21:52:51",
+    "tags"        : [],
+    "draft"       : true,
+    "image"       : "",
+    "thumbnail"   : "",
+    "description" : "Another look at this ubiquitous serialization …",
+    "categories"  : "data"
+  }
+
+[JSON](https://www.json.org/) needs no introduction among most programmers.
+Since its “discovery” by Douglas Crockford in the early 2000s it has become
+one of the most popular data serialization formats yet created, having
+effectively supplanted XML as the web’s “go-to” format for representing
+structured data in APIs.
+
+For all its popularity, though, JSON carries a number of liabilities that
+make it less than ideal in some contexts where it nonetheless enjoys
+widespread use. This article will discuss some of these and propose
+alternative formats that I think deserve your attention.
+
+
+# 1. UTF-8
+
+JSON’s chief liability is its UTF-8 restriction: all JSON documents must
+be UTF-8, from start to finish. This means any application that wants to
+exchange binary data either cannot use JSON, or—perhaps more commonly—uses
```

Comment:
Have you seen [RFC 4627](https://www.rfc-editor.org/rfc/rfc4627.html) ?

> JSON text SHALL be encoded in Unicode.  The default encoding is
   UTF-8.
    Since the first two characters of a JSON text will always be ASCII
    characters [RFC0020], it is possible to determine whether an octet
    stream is UTF-8, UTF-16 (BE or LE), or UTF-32 (BE or LE) by looking
    at the pattern of nulls in the first four octets.

Reply:
This RFC was superseded by RFC 8259